### PR TITLE
Update LSP Packages

### PR DESCRIPTION
**Pull request by DavidKutu**
For #5310

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [x] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [x] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/test/datascience/reactHelpers.ts [hunk 1]
```diff
@@ -414,7 +414,13 @@ export function setUpDomEnvironment() {
             content += 'export function getCSSBasedConfiguration() { return CSSBasedConfiguration.INSTANCE; };\n';
             mod._compile(content, filename);
         } else {
-            _oldLoader(mod, filename);
+            try {
+                _oldLoader(mod, filename);
+            } catch (e) {
```
Comment (DavidKutu) [inline src/test/datascience/reactHelpers.ts:419]: The new LS Packages have issues when loading them to the DOM for tests.
They're not needed so I let them fail. Let me know if we want them loaded for that.

**Review by joyceerhl — APPROVED** [review]

**Review by rchiodo — APPROVED** [review]

File: src/test/datascience/reactHelpers.ts [hunk 2]
```diff
@@ -414,7 +414,13 @@ export function setUpDomEnvironment() {
...
+                console.error(e);
+                console.error(mod);
+                console.error(filename);
```
Comment (DonJayamanne) [inline src/test/datascience/reactHelpers.ts:422]:
```suggestion
                console.error(`Failed to load module for tests ${mod} in ${filename}`, e);
```

File: src/test/datascience/reactHelpers.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/test/datascience/reactHelpers.ts:419]: Hmm, what if something else fails? can we instead check if `e` is something related to LS package and then ignore it, and if its something else we re-throw the error.

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
I'd prefer to check the error before swallowing it (i.e. only ignore LS specific errors & others should be re-thrown)

**Review by IanMatthewHuff — APPROVED** [review]

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**